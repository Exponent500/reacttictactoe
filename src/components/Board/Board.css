.board-row {
    text-align: center;
}

.square-0, .square-1, .square-3, .square-4 {
    border-bottom: 2px solid black;
    border-right: 2px solid black;
}

.square-2, .square-5 {
    border-bottom: 2px solid black;
}

.square-6, .square-7 {
    border-right: 2px solid black;
}
