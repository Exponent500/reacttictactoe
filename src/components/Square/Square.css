.square {
    height: 100px;
    width: 100px;
    line-height: 100px;
    cursor: pointer;
    font-size: 75px;
    /* text-align: center; */
}

.marked {
    cursor: not-allowed;
}