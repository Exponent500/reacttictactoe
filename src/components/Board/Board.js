import React, { Component } from 'react';
import Square from '../Square/Square';
import Grid from '@material-ui/core/Grid';

import './Board.css';

class Board extends Component {
    renderSquare(index) {
        return (
            <Square
                key={index}
                value={this.props.squares[index]}
                onClick={() => this.props.onClick(index)}></Square>
        )
    }

    render() {
        return (
            <Grid container className="board">
                <Grid item xs={12} className="board-row">
                    <Grid container justify="center">
                        <Grid item className="square-0">
                            {this.renderSquare(0)}
                        </Grid>
                        <Grid item className="square-1">
                            {this.renderSquare(1)}
                        </Grid>
                        <Grid item className="square-2">
                            {this.renderSquare(2)}
                        </Grid>
                    </Grid>
                </Grid>
                <Grid item xs={12} className="board-row">
                    <Grid container justify="center">
                        <Grid item className="square-3">
                            {this.renderSquare(3)}
                        </Grid>
                        <Grid item className="square-4">
                            {this.renderSquare(4)}
                        </Grid>
                        <Grid item className="square-5">
                            {this.renderSquare(5)}
                        </Grid>
                    </Grid>
                </Grid>
                <Grid item xs={12} className="board-row">
                    <Grid container justify="center">
                        <Grid item className="square-6">
                            {this.renderSquare(6)}
                        </Grid>
                        <Grid item className="square-7">
                            {this.renderSquare(7)}
                        </Grid>
                        <Grid item className="square-8">
                            {this.renderSquare(8)}
                        </Grid>
                    </Grid>
                </Grid>
            </Grid>
        );
    }
}

export default Board;