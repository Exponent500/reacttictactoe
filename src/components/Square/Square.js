import React, { Component } from 'react';
import classNames from 'classnames';

import './Square.css';

class Square extends Component {
    render() {
        const squareClass = classNames({
            square: true,
            'marked': this.props.value
        });

        return (
            <div className={squareClass} onClick={this.props.onClick}>
                <div className="square-value">
                    {this.props.value}
                </div>
            </div>
        );
    }
}

export default Square;