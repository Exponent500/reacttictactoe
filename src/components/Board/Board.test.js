import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Board from './Board';
import Square from '../Square/Square';

let wrapped;

const mockSquares = [
    'O', 'O', 'X',
    '0', '0', 'X',
    'X', 'X', '0'
];

beforeEach( () => {
    wrapped = shallow(
        <Board squares={mockSquares}></Board>
    ); 
});

describe('Board', () => {
    it('should render correctly', () => {
        expect(shallowToJson(wrapped)).toMatchSnapshot();
    });

    it('should have 9 squares', () => {
        expect(wrapped.find(Square).length).toEqual(9);
    });

    it('should call onClick event when a square is clicked', () => {
        // create a fake event
        const onClick = jest.fn();
        // using mount because there is a need to simulate a click on the Square component in order
        // to enable the Board component to register that a click event occurred
        const wrapped = mount(<Board squares={mockSquares} onClick={onClick} />);
        wrapped.find('.square-value').first().simulate('click');
        expect(onClick).toBeCalledWith(0);
    });
});