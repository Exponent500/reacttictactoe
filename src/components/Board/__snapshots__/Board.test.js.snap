// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board should render correctly 1`] = `
<WithStyles(Grid)
  className="board"
  container={true}
>
  <WithStyles(Grid)
    className="board-row"
    item={true}
    xs={12}
  >
    <WithStyles(Grid)
      container={true}
      justify="center"
    >
      <WithStyles(Grid)
        className="square-0"
        item={true}
      >
        <Square
          key="0"
          onClick={[Function]}
          value="O"
        />
      </WithStyles(Grid)>
      <WithStyles(Grid)
        className="square-1"
        item={true}
      >
        <Square
          key="1"
          onClick={[Function]}
          value="O"
        />
      </WithStyles(Grid)>
      <WithStyles(Grid)
        className="square-2"
        item={true}
      >
        <Square
          key="2"
          onClick={[Function]}
          value="X"
        />
      </WithStyles(Grid)>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
  <WithStyles(Grid)
    className="board-row"
    item={true}
    xs={12}
  >
    <WithStyles(Grid)
      container={true}
      justify="center"
    >
      <WithStyles(Grid)
        className="square-3"
        item={true}
      >
        <Square
          key="3"
          onClick={[Function]}
          value="0"
        />
      </WithStyles(Grid)>
      <WithStyles(Grid)
        className="square-4"
        item={true}
      >
        <Square
          key="4"
          onClick={[Function]}
          value="0"
        />
      </WithStyles(Grid)>
      <WithStyles(Grid)
        className="square-5"
        item={true}
      >
        <Square
          key="5"
          onClick={[Function]}
          value="X"
        />
      </WithStyles(Grid)>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
  <WithStyles(Grid)
    className="board-row"
    item={true}
    xs={12}
  >
    <WithStyles(Grid)
      container={true}
      justify="center"
    >
      <WithStyles(Grid)
        className="square-6"
        item={true}
      >
        <Square
          key="6"
          onClick={[Function]}
          value="X"
        />
      </WithStyles(Grid)>
      <WithStyles(Grid)
        className="square-7"
        item={true}
      >
        <Square
          key="7"
          onClick={[Function]}
          value="X"
        />
      </WithStyles(Grid)>
      <WithStyles(Grid)
        className="square-8"
        item={true}
      >
        <Square
          key="8"
          onClick={[Function]}
          value="0"
        />
      </WithStyles(Grid)>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</WithStyles(Grid)>
`;
